### DISCUSSION TITLE
Header, menu + classless version

### DISCUSSION THREAD

**Discussion by COil** [picocss/pico#642] [Help] [answered]
Hello,

I used [bareCSS](https://github.com/longsien/BareCSS) before, I liked it very much, but it is quite outdated now.
I'd lie to migrate [MicroSymfony](https://github.com/strangebuzz/MicroSymfony) ([website](https://microsymfony.ovh/)) to pico CSS, but I need a header and a dropdown menu (also the dark mode switcher).
How to do this? I noticed that the classless version does not include by default the dropdown menu.

In fact, the structure would be like the [pico CCS doc](https://picocss.com/docs) : 
* A sticky header
* A dropdown menu with some links
* A footer with some links

Thanks. COil.

**Comment by sysmat** [comment, marked as answer]
- dropdown

```html
<details class="dropdown">
        <summary aria-haspopup="group" role="button" class="contrast">
          Dropdown as a button 3
        </summary>
        <ul role="group">
          <li><a>Action</a></li>
          <li><a>Another action</a></li>
          <li role="list" dir="rtl">
            <a href="#" aria-haspopup="listbox">Dropdown</a>
            <ul role="listbox">
              <li><a>sub action</a></li>
              <li><a>Another sub action</a></li>
              <li><a>Something else here in sub</a></li>
            </ul>
          </li>
          <li><a>Something else here</a></li>
        </ul>
      </details>
```

**Reply by COil** [reply]
Thanks, I will try that. :)

**Reply by dday9** [reply]
Per the technical specifications of the summary element, no role is permitted. See MDN's [technical summary of the summary element](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/summary#technical_summary).

**Comment by sysmat** [comment]
- theme switcher

```html
<html data-theme="dark">

<label>
          <input type="checkbox" role="switch" id="themeSwitch" />
</label>

</html>
```

```˙javascript
document.getElementById('themeSwitch')
    .addEventListener('change', (event) => {
        const htmlElement = document.querySelector("html");
        (event.target.checked) ? htmlElement.removeAttribute('data-theme') : htmlElement.setAttribute('data-theme', 'dark');
    });
```

**Comment by Yohn** [comment]
I like the easy way @sysmat set up for a theme switcher, simple and easy to switch.
Pico also had a demo with a a theme switch example [on CodeSandbox](https://codesandbox.io/embed/4mrnhq?view=Editor+%2B+Preview&module=%2Fjs%2Fmodal.js) with the modal example if you'd like to see that too.

**Reply by sysmat** [reply]
@Yohn thx
